### PR TITLE
Documentation build

### PR DESCRIPTION
**Pull request by dopplershift**
Set up some initial sphinx documentation. I'm not sure what's actually public api, so I've only added: `date2num`, `date2index`, `num2date`, `JulianDayFromDate`. I'll leave it to those more knowledgeable to add to that list. I also made some minor tweaks to the docstrings so that the function signatures show up in the docs properly.

This also adds a job on Travis to build the docs; on success this should also deploy from master automatically. If not, I'll make the requisite tweaks to master directly.

### CODE REVIEW

File: ci/deploy_docs.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -e # exit with nonzero exit code if anything fails
```
Comment (ocefpaf) [inline ci/deploy_docs.sh]: I strongly recommend doctr to simplify the docs deployment.

Comment (dopplershift) [reply ci/deploy_docs.sh]: A pull request to the branch is welcome...but at this point I'm using this same procedure on 3 other projects and I don't have the time to figure out doctr.

**Comment by jhamman** [comment]
Should we merge this and open up additional documentation updates to the rest of us?

**Comment by jswhit** [comment]
Yes please.  Will the html docs be put on github/gh-pages, readthedocs, or somewhere else?

**Comment by jhamman** [comment]
@jswhit - gh-pages for now.

**Comment by dopplershift** [comment]
Alright, merging. Now let's see if I hooked up the upload properly. 😁

**Comment by dopplershift** [comment]
Ok, something slightly off...working on it...

**Comment by dopplershift** [comment]
Ok, docs are up: https://unidata.github.io/netcdftime/

**Pull request merged**